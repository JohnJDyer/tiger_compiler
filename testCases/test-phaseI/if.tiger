let
  var a: int;
  var b: array[10] of int;
in
  if a=b[1] then
    a:=b;
  endif;
end