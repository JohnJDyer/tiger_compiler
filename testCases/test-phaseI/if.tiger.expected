main:
	array_load, $t0, b, 1
	assign, $t1, 1, 
	breq, a, $t0, compare_label0
	assign, $t1, 0, 
compare_label0:
	breq, $t1, 0, if_label1
	assign, a, b, 
if_label1:
	return, , , 
